protocols/relay/src/v2.rs

```diff
@@ -42,48 +42,53 @@ pub mod client {
 
         #[deprecated(
             since = "0.15.0",
-            note = "Use libp2p_relay::client::Transport instead."
+            note = "ClientTransport has been renamed to Transport
+            and is now private as it shouldn't be required by end users."
         )]
-        pub type ClientTransport = crate::client::Transport;
+        pub type ClientTransport = crate::priv_client::transport::Transport;
 
         #[deprecated(
             since = "0.15.0",
-            note = "Use libp2p_relay::client::transport::Listener instead."
+            note = "RelayListener has been renamed to Listener
+            and is now private as it shouldn't be required by end users."
         )]
-        pub type RelayListener = crate::client::transport::Listener;
+        pub type RelayListener = crate::priv_client::transport::Listener;
 
         #[deprecated(
             since = "0.15.0",
-            note = "Use libp2p_relay::transport::RelayedDial instead."
+            note = "RelayedDial is now private as it shouldn't be required by end users."
```

Comment:
`RelayedDial` is part of the public API in any case because of `Transport::Dial`. However, it is itself a type alias that we can simply remove (in the future). We should deprecate this and tell users to simply write out the type themselves.